pkg/api/validate/dynamic/dynamic.go

```diff
@@ -389,6 +389,10 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 			}
 		}
 
+		if ss.SubnetPropertiesFormat.ProvisioningState != "Succeeded" {
```

Comment:
Instead of using the hardcoded string "Succeeded", what about using the [ProvisioningState](https://github.com/Azure/azure-sdk-for-go/blob/main/services/network/mgmt/2020-08-01/network/enums.go#L1896) provided by Azure. 

In fact, we already import the network package on top of the file on line 14:
```go
mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
```
so you just need to change your if with one of the two options:
```go
if ss.SubnetPropertiesFormat.ProvisioningState != mgmtnetwork.ProvisioningState(api.ProvisioningStateSucceeded) {
```
or
```go
if ss.SubnetPropertiesFormat.ProvisioningState != mgmtnetwork.Succeeded {
```
?
I would prefer the second one as does not need any type casting.

Reply:
Oh cool, fixed it!